### PR TITLE
dependencies: last_updated and project_desc metadata.

### PR DESCRIPTION
**Pull request by htuch**
* Added mandatory last_updated metadata to track when a dependency was
  last updated. Backfilled existing dependencies based primarily on
  when the dependency was released, and in some cases based on Envoy
  git history.

* Added project_desc metadata to provide a human understandable
  statement about projects.

* Updated CPEs using the latest CPE database. There were a number of
  dataplane components with missing CPEs.

* Corrected use_category for a number of components.

Risk level: Low
Testing: Build, dashboard inspection.

Part of #12673.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -100,206 +111,240 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         strip_prefix = "gperftools-{version}",
         urls = ["https://github.com/gperftools/gperftools/releases/download/gperftools-{version}/gperftools-{version}.tar.gz"],
-        use_category = ["test"],
+        last_updated = "2020-07-06",
+        use_category = ["dataplane", "controlplane"],
+        cpe = "cpe:2.3:a:gperftools_project:gperftools:*",
     ),
     com_github_grpc_grpc = dict(
         project_name = "gRPC",
+        project_desc = "Google gRPC C core library",
```
Comment (moderation) [inline bazel/repository_locations.bzl:250]: With gRPC in it's own Github org / repo now you could probably remove 'Google'

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -37,29 +37,35 @@ USE_CATEGORIES = [
...
-# Components with these use categories are not required to specify the 'cpe' annotation.
+# Components with these use categories are not required to specify the 'cpe'
+# and 'last_updated' annotation.
 USE_CATEGORIES_WITH_CPE_OPTIONAL = ["build", "other", "test"]
 
 DEPENDENCY_REPOSITORIES_SPEC = dict(
     bazel_compdb = dict(
-        project_name = "bazil-compilation-database",
+        project_name = "bazel-compilation-database",
+        project_desc = "clang JSON compilation database support for Bazel",
```
Comment (moderation) [inline bazel/repository_locations.bzl:47]: Clang

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -68,28 +74,33 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         sha256 = "0dc8ce5eb645ae069ce710c1010975456f723ffd4fc788a03dacfcd0647b05b9",
         strip_prefix = "envoy-build-tools-{version}",
-        # 2020-08-21
         urls = ["https://github.com/envoyproxy/envoy-build-tools/archive/{version}.tar.gz"],
+        last_updated = "2020-08-21",
         use_category = ["build"],
     ),
     boringssl = dict(
         project_name = "BoringSSL",
+        project_desc = "Minimal OSS fork",
```
Comment (moderation) [inline bazel/repository_locations.bzl:103]: Minimal OSS fork of OpenSSL

File: bazel/repository_locations.bzl [hunk 4]
```diff
@@ -100,206 +111,240 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         sha256 = "bbc8f020f4e85ec029b047fab939b8c81f3d67254b5c724e1003a2bc49ddd123",
         strip_prefix = "grpc-{version}",
         urls = ["https://github.com/grpc/grpc/archive/{version}.tar.gz"],
         use_category = ["dataplane", "controlplane"],
+        last_updated = "2020-02-11",
         cpe = "cpe:2.3:a:grpc:grpc:*",
     ),
     com_github_luajit_luajit = dict(
         project_name = "LuaJIT",
+        project_desc = "Just-In-Time Compiler for Lua",
```
Comment (moderation) [inline bazel/repository_locations.bzl:264]: compiler

File: bazel/repository_locations.bzl [hunk 5]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         sha256 = "a2faafbc402394df0fa94602df4b5e4befd734aad6bb55dfef46f62fcaf1090b",
         strip_prefix = "rapidjson-{version}",
         urls = ["https://github.com/Tencent/rapidjson/archive/{version}.tar.gz"],
         use_category = ["dataplane"],
+        last_updated = "2019-12-02",
         cpe = "cpe:2.3:a:tencent:rapidjson:*",
     ),
     com_github_twitter_common_lang = dict(
         project_name = "twitter.common.lang (Thrift)",
+        project_desc = "twitter.common python language and compatibility facilities.",
```
Comment (moderation) [inline bazel/repository_locations.bzl:459]: Remove period

File: bazel/repository_locations.bzl [hunk 6]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         strip_prefix = "twitter.common.lang-{version}/src",
         urls = ["https://files.pythonhosted.org/packages/08/bc/d6409a813a9dccd4920a6262eb6e5889e90381453a5f58938ba4cf1d9420/twitter.common.lang-{version}.tar.gz"],
-        use_category = ["dataplane"],
-        cpe = "N/A",
+        last_updated = "2018-06-26",
+        use_category = ["test"],
     ),
     com_github_twitter_common_rpc = dict(
         project_name = "twitter.common.rpc (Thrift)",
+        project_desc = "twitter.common thrift helpers including finagle and ssl transports",
```
Comment (moderation) [inline bazel/repository_locations.bzl:470]: Thrift, Finagle, SSL

File: bazel/repository_locations.bzl [hunk 7]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         strip_prefix = "twitter.common.rpc-{version}/src",
         urls = ["https://files.pythonhosted.org/packages/be/97/f5f701b703d0f25fbf148992cd58d55b4d08d3db785aad209255ee67e2d0/twitter.common.rpc-{version}.tar.gz"],
-        use_category = ["dataplane"],
-        cpe = "N/A",
+        last_updated = "2018-06-26",
+        use_category = ["test"],
     ),
     com_github_twitter_common_finagle_thrift = dict(
         project_name = "twitter.common.finagle-thrift",
+        project_desc = "twitter.common thrift stubs for zipkin rpc tracing support in finagle",
```
Comment (moderation) [inline bazel/repository_locations.bzl:481]: Thrift, Zipkin, RPC, Finagle

File: bazel/repository_locations.bzl [hunk 8]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         strip_prefix = "opencensus-cpp-{version}",
         urls = ["https://github.com/census-instrumentation/opencensus-cpp/archive/{version}.tar.gz"],
         use_category = ["observability"],
+        last_updated = "2020-06-01",
         cpe = "N/A",
     ),
+    # This should be removed, see https://github.com/envoyproxy/envoy/issues/11816.
     com_github_curl = dict(
         project_name = "curl",
+        project_desc = "Library for transferring data with URLs",
```
Comment (moderation) [inline bazel/repository_locations.bzl:597]: Library for transferring data with URLs (OpenCensus tracing dependency)

File: bazel/repository_locations.bzl [hunk 9]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         sha256 = "ef404643d7da6854b76b9fb9950a79a1acbd037b7a26f02c585ac379b0f7dee1",
         urls = ["https://storage.googleapis.com/envoyproxy-wee8/wee8-{version}.tar.gz"],
         use_category = ["dataplane"],
-        cpe = "N/A",
+        last_updated = "2020-08-31",
+        cpe = "cpe:2.3:a:google:v8:*",
     ),
     com_googlesource_quiche = dict(
         project_name = "QUICHE",
+        project_desc = "QUICHE (QUIC, Http/2, Etc) is Google‘s implementation of QUIC and related protocols",
```
Comment (moderation) [inline bazel/repository_locations.bzl:622]: HTTP/2

File: bazel/repository_locations.bzl [hunk 10]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         use_category = ["dataplane"],
+        last_updated = "2020-08-05",
         cpe = "N/A",
     ),
     com_google_cel_cpp = dict(
         project_name = "Common Expression Language C++",
-        project_url = "https://opensource.google/projects/cel",
-        # 2020-07-14
+        project_desc = "Common Expression Language C++",
+        project_url = "CEL C++ library",
```
Comment (moderation) [inline bazel/repository_locations.bzl:647]: Revert to actual URL - https://opensource.google/projects/cel

File: bazel/repository_locations.bzl [hunk 11]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         urls = ["https://storage.googleapis.com/quiche-envoy-integration/googleurl_{version}.tar.gz"],
         use_category = ["dataplane"],
+        last_updated = "2020-08-05",
         cpe = "N/A",
     ),
     com_google_cel_cpp = dict(
         project_name = "Common Expression Language C++",
-        project_url = "https://opensource.google/projects/cel",
-        # 2020-07-14
+        project_desc = "Common Expression Language C++",
```
Comment (moderation) [inline bazel/repository_locations.bzl:646]: Common Expression Language (CEL) C++ library

File: bazel/repository_locations.bzl [hunk 12]
```diff
@@ -311,375 +356,427 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
         sha256 = "494d3f81156b92bac640c26000497fbf3a7b1bc35f9789594280450c6e5d8129",
         strip_prefix = "proxy-wasm-cpp-host-{version}",
         urls = ["https://github.com/proxy-wasm/proxy-wasm-cpp-host/archive/{version}.tar.gz"],
         use_category = ["dataplane"],
+        last_updated = "2020-07-29",
         cpe = "N/A",
     ),
     emscripten_toolchain = dict(
         project_name = "Emscripten SDK",
+        project_desc = "Emscripten SDK (use by Wasm)",
```
Comment (moderation) [inline bazel/repository_locations.bzl:779]: used for Wasm

**Review by yanavlasov — APPROVED** [review]

**Comment by moderation** [comment]
LGTM

**Pull request merged**